tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/statements/ifelse/TypeGuardTest.java

```diff
@@ -167,9 +167,9 @@ public void testTypeGuardSemanticsNegative() {
         BAssertUtil.validateError(negativeResult, i++, "incompatible types: expected 'int', found " +
                 "'(int|string|boolean)'", 293, 25);
         BAssertUtil.validateError(negativeResult, i++, "incompatible types: expected 'int', found " +
-                "'(int|string|boolean)'", 296, 29);
+                "'(int|string)'", 296, 29);
```

Comment:
Is this correct? Shouldn't it be reset to the original type of the variable? For example, in https://github.com/ballerina-platform/ballerina-lang/blob/522385842fc3e961aa99e0dac090f0e193f208c9/tests/jballerina-unit-test/src/test/resources/test-src/statements/ifelse/type-guard-semantics-negative.bal#L292 we could set a boolean value to `x` right? In which case the error message isn't quite right.

Reply:
The following compiles with these changes and results in an incorrect value at runtime.

```ballerina
public function main() {
    int|string|boolean x = 8;
    int|string|boolean y = 8;

    if (x is int|string) {
        int|string a = x;
        if (x is int) {
            int b = x;
            if (x > 5) {
                int c = x;
                x = true;

                if (y is int) {
                    int|string e = x;
                    panic error(string `e's defined type: int|string, actual type: ${(typeof e).toString()}, value: ${e}`);
                }
            }
        }
    }
}
```

```cmd
e's defined type: int|string, actual type: typedesc boolean, value: true
```

Reply:
> Is this correct? Shouldn't it be reset to the original type of the variable? For example, in
> 
> https://github.com/ballerina-platform/ballerina-lang/blob/522385842fc3e961aa99e0dac090f0e193f208c9/tests/jballerina-unit-test/src/test/resources/test-src/statements/ifelse/type-guard-semantics-negative.bal#L292
> 
> we could set a boolean value to `x` right? In which case the error message isn't quite right.

But in this case we have set a string to x right? so it is within the scope starting from line 285. So do we still need to reset back to the original type at line no 282?